### PR TITLE
grpcproxy: stop watchers in watch groups

### PR DESCRIPTION
**Pull request by xiang90**
/cc @heyitsanthony

### CODE REVIEW

File: proxy/grpcproxy/watch.go [hunk 1]
```diff
@@ -203,6 +215,11 @@ func (sws *serverWatchStream) maybeCoalesceWatcher(ws watcherSingle) bool {
 	rid := receiverID{streamID: sws.id, watcherID: ws.w.id}
 	if sws.groups.maybeJoinWatcherSingle(rid, ws) {
 		delete(sws.singles, ws.w.id)
+		// do not add new watchers when stream is closing
+		if sws.inGroups == nil {
```
Comment (heyitsanthony) [inline proxy/grpcproxy/watch.go]: isn't this functionally the same as never setting `sws.inGroups = nil` in the `sws.close` path?

Comment (xiang90) [inline proxy/grpcproxy/watch.go]: we close the watcher async in routines. the watchers can access this map after we release the lock at line 119.

**Comment by xiang90** [comment]
@heyitsanthony PTAL.

**Comment by heyitsanthony** [comment]
lgtm

**Pull request merged**